modules/okhttp/src/main/scala/coursier/cache/protocol/HttpHandler.scala

```diff
@@ -1,17 +1,51 @@
 package coursier.cache.protocol
 
 import java.net.{URLStreamHandler, URLStreamHandlerFactory}
+import java.util.concurrent.TimeUnit
 
 import com.squareup.okhttp.{OkHttpClient, OkUrlFactory}
 
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Contains the default values for the OK Http client.
+  *
+  * The default client can be configured using properties which are first checked from the system properties, then environment variables:
+  *
+  * - COURSIER_CONN_TIMEOUT_SECONDS : the http connection timeout, in seconds
+  * - COURSIER_READ_TIMEOUT_SECONDS : the http read timeout, in seconds
+  *
+  */
 object HttpHandler {
-  lazy val okHttpClient = new OkHttpClient
-  lazy val okHttpFactory = new OkUrlFactory(okHttpClient)
+
+  private def durationInSecondsFromEnv(key : String): Option[FiniteDuration] = {
+    import concurrent.duration._
+    sys.props.get(key).orElse(sys.env.get(key)).map(_.toInt.seconds)
+  }
+
+  def newClient() = {
+    val client = new OkHttpClient
+
+    durationInSecondsFromEnv("COURSIER_CONN_TIMEOUT_SECONDS").foreach { timeoutSeconds =>
```

Comment:
Couldn't this be just `COURSIER_CONN_TIMEOUT` and `COURSIER_READ_TIMEOUT`, then parsed with `scala.concurrent.duration.Duration`? This would put these variables in line with `COURSIER_TTL`.

Also, to keep things line with ttl too, couldn't the Java property be read as e.g. `coursier.conn.timeout` and `coursier.read.timeout`, and only as `COURSIER_CONN_TIMEOUT` and `COURSIER_READ_TIMEOUT` from the environment?